### ISSUE TITLE
Validation errors should be more descriptive

### ISSUE THREAD

**Issue by rdodev** [heptio/ark#298]
This issue is somewhat related to #297 

After scheduling a restore for a non-existent backup, the following validation errors show in the `ark restore describe restore-name`

```
Validation errors:  Error retrieving backup: rpc error: code = Unknown desc = error getting object threefiddy/ark-backup.json: NoSuchKey: The specified key does not exist.
                    status code: 404, request id: 0A75325530AC5C4C, host id: 0e8AIFDYNnFoIttIMSZMe5w+QRXRDmFkp2dPKzuKxjuLWkxv4ls33A8ciAlSz4y24z66hufAu+0=
```
That error isn't particularly descriptive or expressive. All a user could _guess_ from it is that the backup isn't present in the object storage. No mention of what could be causing this error and how to fix it. 

In this contrived example would be trivial for the user to realize they don't have a backup with that name; however, think of a common case that a user could have dozens of similarly named backups 

```
nginx-front-end2018-02-01
nginx-front-end2018-02-02
nginx-front-end2018-02-03
nginx-front-end2018-02-04
nginx-front-end2018-02-05
```

And requesting a restore as follows: `ark restore create nginx-fornt-end2018-02-01`

In short http error codes aren't super helpful, ark should have human-understandable errors.

**Comment by ncdc** [comment]
I'd like to tackle specific error messages/locations that need fixing. That will be easier than trying to audit and improve the entire code base.

For something like this, I think we'll need to make each cloud provider translate from its "no such key" error to an Ark "NotFound" error. For AWS for example, we would modify

https://github.com/heptio/ark/blob/11f2be6254c94f510043c3485f8ed1ee55a1d12b/pkg/cloudprovider/aws/object_store.go#L123-L124

to do something like this:
```go
aerr, ok := err.(awserr.Error)
if ok && aerr.Code() == s3.ErrCodeNoSuchKey {
  return arkerrors.NewBlobNotFound(bucket, key)
}
```

and then we'd maybe modify this

https://github.com/heptio/ark/blob/11f2be6254c94f510043c3485f8ed1ee55a1d12b/pkg/cloudprovider/backup_service.go#L207-L212

to translate the BlobNotFound error to a BackupNotFound error.

**Comment by ncdc** [comment]
Going to close this as it's pretty generic. We can open individual issues for specific things that need fixing.
